test/src/main/java/org/apache/accumulo/test/ComprehensiveIT.java

```diff
@@ -133,6 +152,49 @@ public static void teardown() {
     SharedMiniClusterBase.stopMiniCluster();
   }
 
+  @Test
+  public void testEventualScan() throws Exception {
+    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
+
+      String table = getUniqueNames(1)[0];
+      client.tableOperations().create(table);
+
+      // Start Scan Server
+      getCluster().getClusterControl().start(ServerType.SCAN_SERVER);
+
+      // generate 100 rows of data
+      write(client, table, generateMutations(0, 100, tr -> true));
+
+      // need flush table so that eventual scan can see it.
+      client.tableOperations().flush(table, null, null, true);
```

Comment:
I'm very confused by this approach, and I don't think it's correct. First, it uses `System.currentTimeMillis()` to compute an elapsed time, which we should *never* be doing unless we're estimating elapsed time between two different processes and have no other more authoritative source for time than the system clock.

But more importantly, the time elapsed prior to the conclusion of the flush command doesn't count at all, and there's no point in tracking it. The only time that counts is the time immediately following the conclusion of the flush, because that's the earliest possible time we are guaranteed that the data is written and the flushed file now appears in the metadata for the scan server to see. It does not matter how long the time has elapsed during the flush call... only how much time has elapsed after.

Let's say that the scan server refreshed *immediately* prior to the flush finishing. The test has to wait for the scan server to complete it's *full* 5 second cycle to be guaranteed that the scan server will pick up the new file. But, if the scan server was busy having just finished up a refresh, but its refresh clock hasn't been reset yet, it might take a little bit longer. So, the concept of starting the clock in this test prior to the flush, and only waiting a "remaining" amount doesn't make any sense. You have to fully complete the flush (using wait), and *then* you have to wait at least the full cache refresh time for the scan server, and then maybe a little bit more, before you can guarantee the scan server will pick up the changes.

The only way I can think of shortening the time is to try to verify the data right away, and if it fails and it's less than 5 seconds, wait a tiny bit and then loop to try again. If it takes much longer than the 5 seconds, you can time out and assume it's never going to pick up the new file. But that's probably a bit overkill since you're just keeping the system unnecessarily busy for 5 seconds in the hopes that it will refresh sooner. If you're going to do that, it'd be better to just restart the scan server, to guarantee that it has refreshed data.

I suggest one of the following:

1. flush(wait = true); then sleep 6 seconds (5+20%) to nearly guarantee that it is refreshed; verify data, OR
2. stop scan server; flush(wait = true); restart scan server to absolutely guarantee that it is refreshed; verify data

With option 1, the test may be flaky on extremely rare occassions, but good enough. Option 2 is my preferred approach, assuming it's not substantially slower than option 1.